### ISSUE TITLE
`use_go_plugin_auth` causes a validation error when missing

### ISSUE THREAD

**Issue by patriziobrunops** [TykTechnologies/tyk-operator#499]
Ever since I upgraded to latest `tyk-operator` I've been unable to create/udpate api definitions that don't have `use_go_plugin_auth` or when it is set to `false`.

## Expected Behavior
When `use_go_plugin_auth` is not set, should default to `false`

## Current Behavior
When `use_go_plugin_auth` is not set:
```
error: error validating "dev-config.yaml": error validating data: ValidationError(ApiDefinition.spec): missing required field "use_go_plugin_auth" in io.tyk.tyk.v1alpha1.ApiDefinition.spec; if you choose to ignore these errors, turn validation off with --validate=false
```
When `use_go_plugin_auth` is set to `false`
```
The ApiDefinition "remote-payments-receipts-config" is invalid: spec.use_go_plugin_auth: Required value
```

## Possible Solution
Remove `omitempty` from `UseGoPluginAuth`. Remove `use_go_plugin_auth` from required `ApiDefinition.spec` properties in `crds.yaml`.

## Steps to Reproduce

1.
2.
3.

## Your Environment

* OS (e.g. Linux, macOS, Windows):
  * OS version:
* `tyk-operator` version/commit:

**Comment by buraksekili** [comment]
Hi @patriziobrunops - thank you for informing us. 

This issue should be fixed with #500

**Comment by patriziobrunops** [comment]
> Hi @patriziobrunops - thank you for informing us.
> 
> This issue should be fixed with #500

Hi @buraksekili - #500 fixes the manifest validation error, but if you set `use_go_plugin_auth` to `false` you'll get a failure from the Dashboard API, because of `omitempty`.

**Comment by buraksekili** [comment]
@patriziobrunops I tried to reproduce the following error:

```
The ApiDefinition "remote-payments-receipts-config" is invalid: spec.use_go_plugin_auth: Required value
```

Correct me if I am wrong but this error does not belong to Dashboard API but belongs to validation webhooks. This error happens because the `required` field `spec.use_go_plugin_auth` has `omitempty` tag.  Even though CRD expects this field to present, if it is set as `false`, which is the `nil` value for boolean, your YAML file won't be validated because of `omitempty` tag.

So, as in #500, if CRD does not require users to provide the `use_go_plugin_auth` field, you can either set `use_go_plugin_auth` to `false` or not set it at all.

If you have access to the repository, you can try this approach. 
- Delete `use_go_plugin_auth` from helm crds https://github.com/TykTechnologies/tyk-operator/blob/master/helm/crds/crds.yaml#L1697 
- and re-apply crds, as follows:
```bash
kubectl apply -f ./helm/crds
```

This command will update Tyk CRDs and since `use_go_plugin_auth` is deleted from the required field, you are not required to provide it, or setting it to `false` won't cause any problems.

Please let me know if I wasn't clear or if you have further questions.
